tests/node_selector_test.go

```diff
@@ -0,0 +1,160 @@
+
+	k8sv1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libinfra"
+	"kubevirt.io/kubevirt/tests/libkubevirt"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libvmops"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute] Node Selector", decorators.SigCompute, decorators.SigComputeMigrations, decorators.RequiresTwoSchedulableNodes, decorators.VMLiveUpdateRolloutStrategy, Serial, func() {
+	var (
+		virtClient kubecli.KubevirtClient
+	)
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		kv := libkubevirt.GetCurrentKv(virtClient)
+		kv.Spec.Configuration.VMRolloutStrategy = pointer.P(v1.VMRolloutStrategyLiveUpdate)
+		testsuite.UpdateKubeVirtConfigValue(kv.Spec.Configuration)
+	})
+	Context("for Machine Type", func() {
+
+		const unsupportedMachineType = "pc-q35-test1.2.3"
+
+		BeforeEach(func() {
+			By("Verifying that no nodes have the unsupported machine type label")
+			unsupportedLabel := v1.SupportedMachineTypeLabel + unsupportedMachineType
+			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), k8smetav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, node := range nodes.Items {
+				Expect(node.Labels).ToNot(HaveKeyWithValue(unsupportedLabel, "true"))
+				Expect(node.Labels).ToNot(HaveKeyWithValue(unsupportedLabel, "deprecated"))
+			}
+		})
+
+		It("should prevent pod creation for a VMI using an unsupported machine type", func() {
+			vmi := libvmifact.NewAlpineWithTestTooling(libnet.WithMasqueradeNetworking())
+			vmi.Namespace = testsuite.GetTestNamespace(vmi)
+			vmi.Spec.Domain.Machine = &v1.Machine{Type: unsupportedMachineType}
+
+			vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			var podList *k8sv1.PodList
+			Eventually(func() ([]k8sv1.Pod, error) {
+				podList, err = virtClient.CoreV1().Pods(vmi.Namespace).List(context.Background(), k8smetav1.ListOptions{
+					LabelSelector: fmt.Sprintf("%s=%s", v1.CreatedByLabel, string(vmi.UID)),
+				})
+				Expect(err).NotTo(HaveOccurred())
+				return podList.Items, nil
+			}, 10*time.Second, 1*time.Second).ShouldNot(BeEmpty(), "Pod should eventually be created for the VMI")
+			pod := podList.Items[0]
+
+			By("Validating the pod's affinity contains the machine type label")
+			expectPodNodeSelectorContainsMachineType(pod, unsupportedMachineType)
+
+			By("Validating the pod is in Pending state with Unschedulable condition")
+			Expect(pod.Status.Phase).To(Equal(k8sv1.PodPending), "Pod should remain in Pending state due to node affinity issues")
```

Comment:
this doesn't make sure that the pod is stuck in `pending` , but IMO the next validation is enough.